src/nimble.nim

```diff
@@ -677,9 +677,15 @@ proc getPackageByPattern(pattern: string, options: Options): PackageInfo =
       )
     result = getPkgInfoFromFile(skeletonInfo.myPath, options)
 
+proc projName(options: Options): string =
+  if options.action.projName != "":
+    options.action.projName
+  else:
+    os.getCurrentDir().splitPath.tail.toValidPackageName()
+
 proc dump(options: Options) =
   cli.setSuppressMessages(true)
-  let p = getPackageByPattern(options.action.projName, options)
+  let p = getPackageByPattern(projName(options), options)
```

Comment:
Why use that here? `nimble dump` works fine, no?

Reply:
@dom96 , that's a fix for #640 , where dump was hanging on an invalid .nimble file. I'm new to git, and only after this started making PRs from branches dedicated to different issues.

Reply:
Yeah, please separate these next time. I sadly don't understand your fix so I'm not certain enough to accept it, is it something about the package in #640 that causes that bug? I've ran `nimble dump` on master and it worked fine for Nimble itself.

Reply:
nim-syslog's already fixed its .nimble file. I added a comment to #640 with the .nimble that triggered the problem. With that .nimble file, `options.action.projName` was the empty string.

Reply:
What is it about that Nimble file that causes this?

Reply:
I'm not actually sure, but I suspect the name in the file itself.

It'll be a few days before I can get back around to this, but I'll follow up.